app/scripts/controllers/shopping-cart.js

```diff
@@ -15,45 +15,25 @@ angular.module('Volusion.controllers')
 			$scope.cart = {};
 			$scope.cartEmpty = true;
 			$scope.calcSubtotal = 0;
-			$scope.choices = 99;
 			$scope.coupon = {
 				'code' : '',
-				'show' : false
+				'showApply' : false
 			};
 			$scope.couponsEmpty = false;
 			$scope.loading = false;
 			$scope.showGiftOption = false;
 			$scope.visualCue = false;
-			$scope.msgLength = 0;
-
 			$rootScope.isCartOpen = false;
 
 			translate.addParts('shopping-card');
 
 			function updateCart(showSpinner, callback) {
-
-				if(showSpinner === true){
-					$scope.loading = true;
-				}
-
+				$scope.loading = showSpinner ? true : false;
 				vnCart.updateCart()
 					.then(function () {
-						$scope.cart = vnCart.getCart();
-
-						notifications.displayWarnings($scope.cart.warnings);
-						notifications.displayErrors($scope.cart.serviceErrors);
-
-						if ($scope.cart.warnings && $scope.cart.warnings.length > 0 ||
-							$scope.cart.serviceErrors && $scope.cart.serviceErrors.length > 0) {
-							// scroll cart item to the top so this msg will be visible
-							$rootScope.$emit('vnScroll.cart');
-						} else {
-							if (callback !== undefined) {
-								callback();
-							}
+						if (callback !== undefined) {
```

Comment:
Since you are expecting a function, it is safer to add the check `typeof callback === 'function'` to the `if` condition